src/solver/conv_MP_bidirectional_winograd.cpp

```diff
@@ -530,5 +623,193 @@ template struct ConvMPBidirectWinograd<4, 3>;
+                                wino_out.buff_info.size.c * batch_count,
+                                1,
+                                wino_out.buff_info.size.w * wino_out.buff_info.size.h *
+                                    wino_out.buff_info.size.nk);
+
+    // default conv_desc.
+    // pads{0,0}, stride{1,1}, dilation {1, 1}
+    // trans_output_pads = {0, 0},  group_count = gem_batch_count
+    ConvolutionDescriptor conv_desc({0, 0}, {1, 1}, {1, 1}, {0, 0}, batch_count);
+
+    auto dir = conv::Direction::Forward;
+
+    ConvolutionContext transformed_ctx(in, wei, out, conv_desc, dir, 0);
+    transformed_ctx.ExecutionContext::operator=(ctx);
+
+    DEFINE_WORKSPACE_OFFSETS(wino_in.buff_info.total_byte_size,
+                             wino_wei.buff_info.total_byte_size,
+                             wino_out.buff_info.total_byte_size)
+    auto transform_workSpaceSize = wino_in.buff_info.total_byte_size +
+                                   wino_wei.buff_info.total_byte_size +
+                                   wino_out.buff_info.total_byte_size;
+    auto transform_workSpace = static_cast<char*>(ctx.GetBufs().workSpace);
+    auto gemm_workSpaceSize  = ctx.GetBufs().workSpaceSize - transform_workSpaceSize;
+    auto gemm_workSpace      = (transform_workSpace) + transform_workSpaceSize;
+
+    ConvolutionUserBuffers buff(gemm_workSpace, gemm_workSpaceSize);
+
+    buff.SetFwd(transform_workSpace + wino_in_offset,
+                transform_workSpace + wino_wei_offset,
+                transform_workSpace + wino_out_offset);
+
+    transformed_ctx.SetBufs(buff);
+    transformed_ctx.SetupFloats();
+    return transformed_ctx;
+#else
+    (void)ctx;
+    MIOPEN_THROW(miopenStatusBadParm, "ConvMPBidirectWinograd Unsupported ");
+#endif
+}
+
+template <int WinoDataH, int WinoFilterH, int WinoDataW, int WinoFilterW>
+bool ConvMPBidirectWinograd_xdlops<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>::IsApplicable(
+    const ConvolutionContext& ctx) const
+{
+
+    static const int wino_data_tile   = std::max(WinoDataH, WinoDataW);
+    static const int wino_filter_tile = std::max(WinoFilterH, WinoFilterW);
+
+    if(wino_data_tile == 6 && wino_filter_tile == 3)
+        if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_MP_BD_XDLOPS_WINOGRAD_F6X3{}))
+            return false;
+    if(wino_data_tile == 5 && wino_filter_tile == 3)
+        if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_MP_BD_XDLOPS_WINOGRAD_F5X3{}))
+            return false;
+    if(wino_data_tile == 4 && wino_filter_tile == 3)
+        if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_MP_BD_XDLOPS_WINOGRAD_F4X3{}))
+            return false;
+    if(wino_data_tile == 3 && wino_filter_tile == 3)
+        if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_MP_BD_XDLOPS_WINOGRAD_F3X3{}))
+            return false;
+
+    return IsApplicableTransform<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>(ctx) &&
+           ConvHipImplicitGemmForwardV4R4Xdlops().IsApplicable(GetTransformedConvContext(ctx));
```

Comment:
Depends on `ConvHipImplicitGemmForwardV4R4Xdlops`?

Reply:
Yes

Reply:
We need to write a proper solver for it, ```ConvHipImplicitGemmForwardV4R4Xdlops``` is not using the best layout. I would need to know what layout you are using for the transformation kernels.

Reply:
It is necessary to clarify why this is necessary. Please add respective comment.

Please separately comment cases when:
- The `ConvHipImplicitGemmForwardV4R4Xdlops` solver is disabled (e.g. via environment).
- The iGEMM algo is totally disabled.

Reply:
Is this PR using Tensile or ```ConvHipImplicitGemmForwardV4R4Xdlops``` for the GEMM part?

Reply:
@asroy 
>Is this PR using Tensile or ConvHipImplicitGemmForwardV4R4Xdlops for the GEMM part?

ConvHipImplicitGemmForwardV4R4Xdlops 


>We need to write a proper solver for it, ConvHipImplicitGemmForwardV4R4Xdlops is not using the best layout. I would need to know what layout you are using for the transformation kernels.

The original format presented in ConvMPBidirectWinograd is - GCNHW (for data and filter).
To work with ConvHipImplicitGemmForwardV4R4Xdlops the context is converted to NGCHW but N=1, G=G, C=C, H=1, W=H*G*W (for data), and in filter buffer used GNCHW (G=G, N=K, C=C, H=1, W=1). (The filter kernel code was changed specifically for this purpose)

Reply:
@atamazov 

>It is necessary to clarify why this is necessary. Please add respective comment.
>
>Please separately comment cases when:
>
>The ConvHipImplicitGemmForwardV4R4Xdlops solver is disabled (e.g. via environment).

As you can see, there is no such environment variable here
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/3ff4d7fec01888ed18fad69565f703b82afee451/src/solver/conv_hip_implicit_gemm_v4r4_gen_xdlops.cpp#L439-L451

But if conv_hip_implicit_gemm_v4r4_gen_xdlops is disabled, this solver disabled to. (concept)

>The iGEMM algo is totally disabled.

Same principle as with GEMM.  This solver will be enabled.

I don't see any non-obvious points here. The previous solution worked on the same logic.

Reply:
OK